ipns/errors.go

```diff
@@ -4,41 +4,45 @@ import (
+const MaxRecordSize int = 10 << (10 * 1)
 
-// ErrUnrecognizedValidity is returned when an IpnsRecord has an
-// unknown validity type.
-var ErrUnrecognizedValidity = errors.New("unrecognized validity type")
+// ErrExpiredRecord is returned when an IPNS [Record] is invalid due to being expired.
+var ErrExpiredRecord = errors.New("the IPNS record is expired")
 
-// ErrInvalidPath should be returned when an ipns record path
-// is not in a valid format
-var ErrInvalidPath = errors.New("record path invalid")
+// ErrUnrecognizedValidity is returned when an IPNS [Record] has an unknown validity type.
+var ErrUnrecognizedValidity = errors.New("the IPNS record contains an unrecognized validity type")
 
-// ErrSignature should be returned when an ipns record fails
-// signature verification
-var ErrSignature = errors.New("record signature verification failed")
+// ErrInvalidValidity is returned when an IPNS [Record] has a known validity type,
+// but the validity value is invalid.
+var ErrInvalidValidity = errors.New("the IPNS record contains an invalid validity")
 
-// ErrKeyFormat should be returned when an ipns record key is
-// incorrectly formatted (not a peer ID)
-var ErrKeyFormat = errors.New("record key could not be parsed into peer ID")
+// ErrRecordSize is returned when an IPNS [Record] exceeds the maximum size.
+var ErrRecordSize = errors.New("the IPNS record exceeds allowed size limit")
 
-// ErrPublicKeyNotFound should be returned when the public key
-// corresponding to the ipns record path cannot be retrieved
-// from the peer store
-var ErrPublicKeyNotFound = errors.New("public key not found in peer store")
+// ErrDataMissing is returned when an IPNS [Record] is missing the data field.
+var ErrDataMissing = errors.New("the IPNS record is missing the data field")
 
-// ErrPublicKeyMismatch should be returned when the public key embedded in the
-// record doesn't match the expected public key.
-var ErrPublicKeyMismatch = errors.New("public key in record did not match expected pubkey")
+// ErrInvalidRecord is returned when an IPNS [Record] is malformed.
+var ErrInvalidRecord = errors.New("the IPNS record is malformed")
 
-// ErrBadRecord should be returned when an ipns record cannot be unmarshalled
-var ErrBadRecord = errors.New("record could not be unmarshalled")
+// ErrPublicKeyMismatch is return when the public key embedded in an IPNS [Record]
+// does not match the expected public key.
+var ErrPublicKeyMismatch = errors.New("the IPNS record public key does not match the expected public key")
 
-// 10 KiB limit defined in https://github.com/ipfs/specs/pull/319
-const MaxRecordSize int = 10 << (10 * 1)
+// ErrPublicKeyNotFound is returned when the public key is not found.
+var ErrPublicKeyNotFound = errors.New("public key not found")
+
+// ErrInvalidPublicKey is returned when an IPNS [Record] has an invalid public key,
+var ErrInvalidPublicKey = errors.New("the IPNS record public key invalid")
```

Comment:
we use this outside record (when parsing identity CID), so maybe:
```suggestion
var ErrInvalidPublicKey = errors.New("invalid public key")
```